controller/RestDelivery.php

```diff
@@ -137,79 +125,99 @@ public function generateDeferred()
-                        return;
+            $deliveryModelClass = $this->getDeliveryRootClass();
+            $deliveries = $deliveryModelClass->searchInstances($where, ['like' => false, 'recursive' => true]);
+
+            $propertyValues = $_POST;
+            $response = [];
+
+            /** @var \core_kernel_classes_Resource $delivery */
+            foreach ($deliveries as $key => $delivery) {
+                $queueService = $this->getServiceManager()->get(Queue::SERVICE_ID);
+                if ($taskResource = $queueService->getTaskResource($delivery)) {
+                    /** @var Task $task */
+                    $task = $queueService->getTask($taskResource->getUri());
+                    if ($task
+                        && ($task->getStatus() == Task::STATUS_CREATED
+                            || $task->getStatus() == Task::STATUS_RUNNING
+                            || $task->getStatus() == Task::STATUS_STARTED)
+                    ) {
+                        break;
+                    } else if ($task && $task->getStatus() == Task::STATUS_FINISHED) {
+                        /** @var \common_report_Report $report */
+                        $report = $queueService->getReportByLinkedResource($delivery);
+                        if ($report->getType() == Report::TYPE_ERROR) {
+                            break;
+                        }
                     }
                 }
-            }
-            if ($this->hasRequestParameter(self::REST_DELIVERY_PARAMS)) {
-                $propertyValues = $this->getRequestParameter(self::REST_DELIVERY_PARAMS);
-                $propertyValues = json_decode(html_entity_decode($propertyValues), true);
+
                 foreach ($propertyValues as $rdfKey => $rdfValue) {
+                    $rdfKey = \tao_helpers_Uri::decode($rdfKey);
                     $property = $this->getProperty($rdfKey);
                     $delivery->editPropertyValues($property, $rdfValue);
                 }
+
+                $response[] = ['delivery' => $delivery->getUri()];
             }
-            $this->returnSuccess(array('delivery' => $delivery->getUri()));
+            $this->returnSuccess($response);
         }catch (\Exception $e) {
                 $this->returnFailure($e);
             }
     }
 
     /**
-     * Action to find delivery by parameters
+     * Update delivery by parameters
      */
-    public function search()
+    public function updateDeferred()
     {
         try {
-            if ($this->getRequestMethod() !== \Request::HTTP_GET) {
-                throw new \common_exception_NotImplemented('Only get method is accepted to searching delivery');
+            if ($this->getRequestMethod() !== \Request::HTTP_POST) {
+                throw new \common_exception_NotImplemented('Only post method is accepted to updating delivery');
             }
-            $params = $this->getRequestParameters();
+
+            $searchParams = $_GET;
```

Comment:
Do not use all the data from request query. I guess it would be better to have `searchParams` GET parameter with array of parameters encoded into json. It will be more easy to document.